### PR TITLE
Devirtualizer: enable inline an inlinable function into a non-inlinable function

### PR DESCRIPTION
**Pull request by shajrawi**
Follow-up to https://github.com/apple/swift/pull/17327

### CODE REVIEW

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

File: lib/SILOptimizer/Utils/Devirtualize.cpp [hunk 1]
```diff
@@ -515,10 +515,8 @@ bool swift::canDevirtualizeClassMethod(FullApplySite AI,
...
   // We need to disable the  “effectively final” opt if a function is inlinable
-  if (isEffectivelyFinalMethod &&
-      ((F->getResilienceExpansion() == ResilienceExpansion::Minimal) ||
-       AI.getFunction()->getResilienceExpansion() ==
-           ResilienceExpansion::Minimal)) {
+  if (isEffectivelyFinalMethod && AI.getFunction()->getResilienceExpansion() ==
+                                      ResilienceExpansion::Minimal) {
     DEBUG(llvm::dbgs() << "        FAIL: Could not optimize function because "
                           "it is an effectively-final inlinable: "
                        << F->getName() << "\n");
```
Comment (jrose-apple) [inline lib/SILOptimizer/Utils/Devirtualize.cpp]: This should probably be using `AI.getFunction()` too, right?

Comment (shajrawi) [reply lib/SILOptimizer/Utils/Devirtualize.cpp]: Right!

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

**Comment by shajrawi** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**